### DISCUSSION TITLE
Import SVG files without disturbing parent HTML element

### DISCUSSION THREAD

**Discussion by pmwmedia** [solidjs/solid#918] [Q&A]
I use Webpack with `svg-inline-loader` for importing SVG files. These I can insert as `innerHTML`:

```tsx
import icon from "../resources/webapp/maskicon.svg"

function MaskIcon() {
    return <div innerHTML={icon} />
}
```

This works so far. However, I need the SVG file without a boilerplate parent `div`. Is there anyway to insert SVG files without being forced to use a disturbing parent HTML element?

**Comment by mountfx** [comment]
Instead of inserting it via `innerHTML` I think you can just return the svg directly.
Like so: https://playground.solidjs.com/?hash=-1099081994&version=1.3.9

**Reply by edemaine** [reply]
If you can only access the SVG as text, you can use the `template` function from `solid-js` at the top level, and then `cloneNode` within your component. This is actually what the previous example compiles to:

```js
import { render, template } from 'solid-js/web';

const _tmpl$ = /*#__PURE__*/template(`<svg><rect width="400" height="200" fill="gray"></rect></svg>`, 4);

function MaskIcon() {
  return _tmpl$.cloneNode(true);
}
```

Note: The previous example has a bug. You can't put the JSX outside of the function, or else multiple instances of the icon won't work. I've fixed that here.

**Reply by pmwmedia** [reply]
Thank you very much! This is exactly for what I was looking for.
